app/src/main/java/ch/hikemate/app/model/route/HikeRoutesRepositoryOverpass.kt

```diff
@@ -422,4 +446,25 @@ class HikeRoutesRepositoryOverpass(val client: OkHttpClient) : HikeRoutesReposit
       return LatLong(lat, lon)
     }
   }
+
+  /**
+   * The response handler for the Overpass API request.
+   *
+   * @param onHike To be called if a single route is found.
+   * @param noSingleHike To be called if no or multiple routes are found.
+   */
+  private inner class OverpassSingleRouteHandler(
+      val onHike: (HikeRoute) -> Unit,
+      val noSingleHike: (Exception) -> Unit,
+  ) :
+      OverpassResponseHandler(
+          requestedBounds = null,
+          onSuccess = { routes ->
+            when {
+              routes.isEmpty() -> noSingleHike(Exception("No route found"))
+              routes.size != 1 -> noSingleHike(Exception("Multiple routes found"))
+              else -> onHike(routes.first())
+            }
+          },
+          onFailure = { e -> noSingleHike(e) })
```

Comment:
Praise: Never thought of this when having done the initial OverpassRouteHandler. What a great, simple and clean way to do it!!!!

Reply:
Yes the implementation was very easy thanks to how clean and well-thought your code is